# 📊 Análisis Exploratorio de Datos con Python  

## Daniel Mauricio Cárdenas Forero

[LinkedIn](https://www.linkedin.com/in/dcardenasf/) | [Github](https://github.com/dcardenasf)

---

## 🎯 Objetivo:

Realizar un análisis exploratorio de datos, utilizando un conjuno de datos (dataset) en ``Python``


## 🧠 Introducción
El Análisis Exploratorio de Datos (AED o EDA, por sus siglas en inglés) es una etapa fundamental dentro del proceso de ciencia de datos. Su objetivo principal es comprender la estructura, calidad y características generales de un conjunto de datos antes de aplicar modelos estadísticos o de machine learning. En este ejercicio, se utilizará Python y sus bibliotecas más comunes para realizar un análisis exploratorio completo, desde la carga de datos hasta la visualización de patrones y relaciones clave.

## 🛠️ Metodología
Para llevar a cabo el análisis exploratorio, se seguirán los siguientes pasos:

1. **Carga del dataset**
Se utilizará la biblioteca pandas para importar el conjunto de datos desde una fuente local o en línea, y así comenzar su manipulación y análisis.

2. **Revisión inicial de los datos**
Incluye una inspección general de las dimensiones del dataset, tipos de datos, valores nulos, duplicados y primeras filas para tener una visión preliminar.

3. **Limpieza de datos**
En esta etapa se tratarán valores faltantes, datos atípicos (outliers) y conversiones necesarias de tipos de datos para asegurar la calidad del análisis.

4. **Análisis univariado y estadístico descriptivo**
Se explorarán las variables de forma individual utilizando pandas, numpy y seaborn para generar métricas como la media, mediana, desviación estándar, entre otros.

5. **Análisis bivariado y multivariado**
Se analizarán las relaciones entre variables usando gráficos de dispersión, mapas de calor y análisis de correlaciones, con ayuda de matplotlib, seaborn y plotly.

6. **Visualización de datos**
Se utilizarán herramientas gráficas para identificar patrones, distribuciones, relaciones y comportamientos anómalos dentro del conjunto de datos. Esto permite obtener insights valiosos de manera visual y comprensible.

7. **Conclusiones**
Finalmente, se presentarán los hallazgos más relevantes del análisis exploratorio, destacando aspectos interesantes que podrían guiar análisis posteriores o la toma de decisiones.

## 0. Preliminares  

Vamos a realizar la carga de las librerias básicas que se van a utilizar para este ejercicio.

- pandas: Permite la carga, manipulación y limpieza eficiente de datasets tabulares.
- numpy: Complementa el análisis numérico con funciones matemáticas y estadísticas rápidas.
- matplotlib: Proporciona las herramientas básicas para generar gráficos estáticos.
- seaborn: Se basa en matplotlib, pero ofrece una forma más sencilla y atractiva de crear gráficos estadísticos.
- plotly: Permite generar visualizaciones interactivas que mejoran la exploración y comprensión de los datos.

In [ ]:
import pandas as pd        # Manipulación y análisis de datos estructurados
import numpy as np         # Operaciones matemáticas y estadísticas de bajo nivel
import matplotlib.pyplot as plt  # Visualización básica de datos (gráficos de líneas, barras, etc.)
import seaborn as sns      # Visualización avanzada y estética sobre Matplotlib
import plotly.express as px # Visualización interactiva de datos

## 1. Carga del dataset.

En esta sección realizaremos un análisis exploratorio del dataset de ventas globales de videojuegos entre 1980 y 2020. Este conjunto de datos permite realizar distintos tipos de análisis enfocados en las ventas por plataforma, género y región.

A continuación, se describen las columnas disponibles en el dataset:

| Columna | Descripción |
|---|---|
|Rank| Posición del juego según las ventas totales|
|Name| Nombre del videojuego|
|Platform| Consola en la que fue publicado|
|Year| Año de lanzamiento|
|Genre | Género del videojuego |
|Publisher| Distribuidor del videojuego|
|NA_Sales| Ventas en Norteamérica (en millones)|
|EU_Sales| Ventas en Europa (en millones)|
|JP_Sales| Ventas en Japón (en millones)|
|Other_Sales| Ventas en el resto del mundo (en millones)|
|Global_Sales| Ventas totales a nivel mundial (en millones)|

*Fuente: [Video Game Sales](https://www.kaggle.com/datasets/anandshaw2001/video-game-sales?resource=download)*

In [ ]:
path = "https://raw.githubusercontent.com/Dcardenasf/Portafolio/refs/heads/main/Data/Raw/vgsales.csv" # Ruta al archivo CSV
df = pd.read_csv(path)  # Cargar el archivo CSV en un DataFrame de pandas

Rank                       Name Platform    Year         Genre Publisher  \
0     1                 Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2          Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3             Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4          Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5   Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
5     6                     Tetris       GB  1989.0        Puzzle  Nintendo   
6     7      New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
7     8                   Wii Play      Wii  2006.0          Misc  Nintendo   
8     9  New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
9    10                  Duck Hunt      NES  1984.0       Shooter  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37  
5     23.20      2.26      4.22         0.58         30.26  
6     11.38      9.23      6.50         2.90         30.01  
7     14.03      9.20      2.93         2.85         29.02  
8     14.59      7.06      4.70         2.26         28.62  
9     26.93      0.63      0.28         0.47         28.31

# 2. Revisión inicial de los datos  

En esta sección realizaremos una inspección preliminar del dataset. El objetivo es comprender la estructura general de los datos y detectar posibles problemas que debamos tratar antes del análisis.

Las tareas que abordaremos son:

- Visualizar las primeras filas del dataset.

- Identificar el tipo de datos de cada columna.

- Detectar valores nulos o faltantes.

- Buscar registros duplicados.

- Revisar estadísticas descriptivas básicas para variables numéricas.

Estas acciones nos permitirán establecer una base sólida para el análisis posterior

In [12]:
df.head() # Mostrar las primeras filas del DataFrame


Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [14]:
df.info() # Información general del DataFrame, incluyendo el número de entradas y tipos de datos

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


In [15]:
df.isnull().sum() # Contar los valores nulos en cada columna


Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [18]:
df.duplicated().sum() # Comprobar si hay filas duplicadas en el DataFrame


np.int64(0)

Con base en la revisión inicial, podemos concluir que el dataset, contiene 16598 observaciones y 11 columnas con distintos tipos númericos (int64, float64) y categóricos (object).

Se identifican valores nulos en las columnas `Year` y `Publisher`; sin embargo no se encontraron valores duplicados.

Esto nos indica que será necesario tratar los valores nulos antes de continuar con el análisis exploratorio o la visualización de datos.